c/scheme.c

```diff
@@ -1107,20 +1106,37 @@ extern void Sregister_boot_file_fd_region(const char *name,
                                           iptr offset,
                                           iptr len,
                                           int close_after) {
+  struct boot_desc *boot;
+
   check_boot_file_state("Sregister_boot_file_fd_region");
 
-  if (strlen(name) >= BOOT_PATH_MAX) {
-    fprintf(stderr, "boot-file path is too long %s\n", name);
-    S_abnormal_exit();
-  }
+  boot = malloc(sizeof(boot_desc));
 
-  bd[boot_count].fd = fd;
-  bd[boot_count].offset = offset;
-  bd[boot_count].len = len;
-  bd[boot_count].need_check = 1;
-  bd[boot_count].close_after = close_after;
-  strcpy(bd[boot_count].path, name);
-  boot_count += 1;
+  S_fasl_init_fd(&boot->ffo, (ptr)0, fd, FASL_BUFFER_READ_REMAINING, len);
+  boot->offset = offset;
+  boot->len = len;
+  boot->is_fd = 1;
+  boot->need_check = 1;
+  boot->close_after = close_after;
+  add_boot(boot, name);
+}
+
+extern void Sregister_boot_file_bytes(const char *name,
+                                      void *data,
+                                      iptr len) {
+  struct boot_desc *boot;
+
+  check_boot_file_state("Sregister_boot_bytes");
```

Comment:
Shouldn't this be `Sregister_boot_file_bytes`?

Reply:
Yes, thanks.